pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	return k8ssecret.Query(g.client, g.apiReader, log).DeleteForNamespaces(ctx, consts.BootsTrapperInitSecretName, nsList)
+}
+
+// generate gets the necessary info the create the init secret data
+func (g *BootstrapperInitGenerator) generate(ctx context.Context, dk *dynakube.DynaKube) (map[string][]byte, error) {
+	hostMonitoringNodes, err := g.getHostMonitoringNodes(dk)
+	if err != nil {
+		return nil, err
+	}
+
+	secretConfig, err := g.createSecretConfigForDynaKube(ctx, dk, hostMonitoringNodes)
+	if err != nil {
+		return nil, err
+	}
+
+	agCerts, err := dk.ActiveGateTLSCert(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	trustedCAs, err := dk.TrustedCAs(ctx, g.apiReader)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	pmcSecret, err := g.preparePMCSecretForBootstrapper(ctx, *dk)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return map[string][]byte{
+		pmc.InputFileName:        pmcSecret,
+		ca.TrustedCertsInputFile: trustedCAs,
+		ca.AgCertsInputFile:      agCerts,
+		curl.InputFileName:       []byte(strconv.Itoa(secretConfig.InitialConnectRetry)),
+		endpoint.InputFileName:   []byte(secretConfig.ApiToken),
+	}, nil
+}
+
+func (g *BootstrapperInitGenerator) preparePMCSecretForBootstrapper(ctx context.Context, dk dynakube.DynaKube) ([]byte, error) {
+	pmc, err := g.dtClient.GetProcessModuleConfig(ctx, 0)
+	if err != nil {
+		conditions.SetDynatraceApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+		return nil, err
+	}
+
+	tenantToken, err := k8ssecret.GetDataFromSecretName(ctx, g.apiReader, types.NamespacedName{
+		Name:      dk.OneAgent().GetTenantSecret(),
+		Namespace: dk.Namespace,
+	}, connectioninfo.TenantTokenKey, log)
+	if err != nil {
+		conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+		return nil, err
+	}
+
+	pmc = pmc.
+		AddHostGroup(dk.OneAgent().GetHostGroup()).
+		AddConnectionInfo(dk.Status.OneAgent.ConnectionInfoStatus, tenantToken).
+		// set proxy explicitly empty, so old proxy settings get deleted where necessary
+		AddProxy("")
+
+	if dk.NeedsOneAgentProxy() {
+		proxy, err := dk.Proxy(ctx, g.apiReader)
+		if err != nil {
+			conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+			return nil, err
+		}
+
+		pmc.AddProxy(proxy)
+
+		multiCap := capability.NewMultiCapability(&dk)
+		dnsEntry := capability.BuildDNSEntryPointWithoutEnvVars(dk.Name, dk.Namespace, multiCap)
+
+		if dk.FeatureNoProxy() != "" {
+			dnsEntry += "," + dk.FeatureNoProxy()
+		}
+
+		pmc.AddNoProxy(dnsEntry)
+	}
+
+	marshaled, err := json.Marshal(pmc)
+	if err != nil {
+		log.Info("could not marshal process module config")
+
+		return nil, err
+	}
+
+	return marshaled, err
+}
+
+func (g *BootstrapperInitGenerator) createSecretConfigForDynaKube(ctx context.Context, dk *dynakube.DynaKube, hostMonitoringNodes map[string]string) (*startup.SecretConfig, error) {
```

Comment:
why, when you have `generate`?

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/8d9deb9183a43f3c0402f76ec0be8738de41bb50